common/src/main/java/com/fasterxml/jackson/databind/introspect/GuiceInjectableValues.java

```diff
@@ -36,6 +37,17 @@ public Object findInjectableValue(
       Object valueId, DeserializationContext ctxt, BeanProperty forProperty, Object beanInstance
   )
   {
-    return injector.getInstance((Key) valueId);
+    if(Key.class.isAssignableFrom(valueId.getClass())){
+      return injector.getInstance((Key) valueId);
+    }
+    if(String.class.isAssignableFrom(valueId.getClass())){
+      try {
+        return injector.getInstance(getClass().getClassLoader().loadClass((String)valueId));
```

Comment:
I think this was leftover from some of my advanced Jackson handling tests, which didn't end up working out. Will revert